addons/mail/models/models.py

```diff
@@ -54,7 +54,7 @@ def _mail_track(self, tracked_fields, initial):
                 tracking = self.env['mail.tracking.value'].create_tracking_values(initial_value, new_value, col_name, col_info, tracking_sequence, self._name)
                 if tracking:
                     if tracking['field_type'] == 'monetary':
-                        tracking['currency_id'] = getattr(self, col_info.get('currency_field', ''), self.company_id.currency_id).id
+                        tracking['currency_id'] = getattr(self, col_info.get('currency_field')).id
```

Comment:
```suggestion
                        tracking['currency_id'] = self[col_info['currency_field']].id
```
Since the `currency_field` is mandatory we can use a non-optional field lookup (as that would return `None` which would trigger an error during the `getattr`, here we get a clearer `KeyError` with the key name), and given `*currency_field` is a field, we can just look it up using the mapping syntax.

That way there's no need for the `getattr` at all.